riscv/src/compiler.rs

```diff
@@ -904,6 +904,27 @@ fn process_instruction(instr: &str, args: &[Argument]) -> Vec<String> {
             let (rd, r1, r2) = rrr(args);
             only_if_no_write_to_zero(format!("tmp1, {rd} <== mul({r1}, {r2});"), rd)
         }
+        "mulhsu" => {
+            let (rd, r1, r2) = rrr(args);
+            only_if_no_write_to_zero_vec(
+                vec![
+                    format!("tmp1 <== to_signed({r1});"),
+                    // tmp2 is 1 if tmp1 is non-negative
+                    "tmp2 <== is_positive(tmp1 + 1);".into(),
+                    // If negative, convert to positive
+                    "skip_if_zero 0, tmp2;".into(),
+                    "tmp1 <=X= 0 - tmp1;".into(),
+                    format!("tmp1, {rd} <== mul(tmp1, {r2});"),
```

Comment:
Is there a test with tmp1 being the most negative number? I'm asking because if you negate it, it does not fit 16 bits any more and this could create problems when using the `mul` operation.

Reply:
Assuming you meant 32 bits, it does fit. The most negative number is -2**31, the mul operands range in 0 to (2**32-1), so 2**31 is within range.

Reply:
And is there a test?

Reply:
`test_7` covers the most negative value